staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/controller/controller.go

```diff
@@ -163,7 +164,7 @@ func (d *DynamicKMSEncryptionConfigContent) processNextWorkItem(serverCtx contex
 	ctx, closeTransformers := context.WithCancel(serverCtx)
 
 	defer func() {
-		// TODO: increment success metric when updatedEffectiveConfig=true
+		metrics.RecordEncryptionConfigHotReloadSuccess()
```

Comment:
The `err != nil` check is done later, so this would update the success metric even if there is a failure?

Reply:
update the logic